board/download_dependencies.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+#
+# This script download all dependecies for mlpack and cross-compile them against
+# required architecture.
+# 
+
+version=$1
+version_underscore=$2
+arch=$3
+binary=$4
+
+# echo "Download and build boost as a static lbrary"
+# wget --no-check-certificate \
+#       "https://dl.bintray.com/boostorg/release/$version/source/boost_$version_underscore.tar.gz" && \
+#     tar xvzf boost_$version_underscore.tar.gz && \
+#     rm -f boost_$version_underscore.tar.gz && \
+#     cd boost_$version_underscore && \
+#     cp tools/build/example/user-config.jam project-config.jam && \
+#     echo 'using gcc : arm : aarch64-linux-gnu-g++ ;' >> project-config.jam &&\
+#     b2 --build=$PWD/build toolset=gcc-arm link=static cxxflags=-fPIC && \
+#     -with-libraries=math,program_options,serialization,test && \
+#     cd ../../ && \
+
+
+# echo "Downlaod and build OpenBLAS as a static library"
+# wget --no-check-certificate \
+#       "https://github.com/xianyi/OpenBLAS/archive/v0.3.9.tar.gz" && \
+#       tar xvzf v0.3.9.tar.gz && \
+#       rm -f v0.3.9.tar.gz && \
+#       cd OpenBLAS-0.3.9 && \
+#       make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=aarch64-linux-gnu-gcc NOFORTRAN=1 libs &&\
+#       cd ../ &&\
+
+# echo "Download and build armadillo as a header"
+# wget --no-check-certificate \
+#     http://sourceforge.net/projects/arma/files/armadillo-9.880.1.tar.xz && \
+#     tar xvf armadillo-9.880.1.tar.xz && \
+#     rm -f armadillo-9.880.1.tar.xz && \
+#     cd  armadillo-9.880.1
+
+echo "Download and build armadillo as a header"
+wget --no-check-certificate \
+    https://github.com/Reference-LAPACK/lapack/archive/v3.9.0.tar.gz && \
+    tar xvf v3.9.0.tar.gz && \
+    rm -f v3.9.0.tar.xz && \
+    cp fortran/make.inc lapack-3.9.0
+    cd lapack-3.9.0
+    make lapacklib -j32
```

Comment:
I'm not sure how feasible it will be on all devices, but it may be better (and easier) to use OpenBLAS.  If I remember right, OpenBLAS won't need a FORTRAN compiler (it's all C/C++), and I *think* it's portable to other architectures too.  However I'm not 100% sure.  In any case, I think OpenBLAS will perform better than the reference BLAS/LAPACK implementation too (and for systems with multiple cores, it is parallelized too).  Maybe I overlooked something that makes it infeasible, if so, let me know. :)

Reply:
@rcurtin Yes, thanks for reviewing I was submerged in this issue, I did not notice your message.
I had left a comment in the last commit message.
In fact, I have added OpenBlast without Fortran, and the linker complained of undefined references. It seems that we need some sort of LAPACK
Finally, I had to use CLAPACK, this one was a little complicated to integrate, but it worked finally.

Reply:
Hmm, maybe that's true, I guess I'm not sure.  I know that the Julia project builds OpenBLAS for their binary packaging system (which includes ARM): https://github.com/JuliaPackaging/Yggdrasil/blob/master/O/OpenBLAS/common.jl#L25

Not sure how helpful that link is, but maybe some of the things about how they did it could be helpful.  At least my experiences into digging around into getting BLAS/LAPACK replacements to compile on weird architectures have been... confusing.  Probably because I haven't spent enough time with it. :)